vendor/bat-native-ads/src/bat/ads/internal/ads/serving/choose/eligible_ads_predictor_util.h

```diff
@@ -159,6 +161,37 @@ CreativeAdPredictorMap<T> ComputePredictorFeaturesAndScores(
   return creative_ad_predictors_with_features;
 }
 
+template <typename T>
+std::vector<int> ComputeVoteRegistry(
+    const std::vector<T>& creative_ads,
+    const TextEmbeddingHtmlEventList& text_embedding_html_events) {
+  DCHECK(!creative_ads.empty());
+
+  std::vector<int> vote_registry;
+  vote_registry.assign(creative_ads.size(), 0);
+
+  for (const auto& text_embedding : text_embedding_html_events) {
+    int max_index = 0;
+    float max_similarity = 0.0;
+    for (const auto& creative_ad : creative_ads) {
```

Comment:
could we do just an indexed for loop here? If so, we don't need to calculate max_index (which gives square complexity):
```
        max_index =
             std::find(creative_ads.cbegin(), creative_ads.cend(), creative_ad) -
             creative_ads.begin();
```